lib/util/npm-util.js

```diff
@@ -34,6 +36,8 @@ function findPackageJson(startDir) {
         if (!fs.existsSync(pkgFile) || !fs.statSync(pkgFile).isFile()) {
             dir = path.join(dir, "..");
             continue;
+        } else if (fs.existsSync(path.join(dir, "yarn.lock"))) {
```

Comment:
You mean, that either the project packages are managed by yarn (yarn.lock is present), eslint should attempt to install packages via NPM if it can't find yarn installed? I think it's a bad idea to use NPM on any project that already using yarn - it will leave broken package lock file and create a lot of confusion between collaborators. So, execution failure will be a better result, preventing further problems.

If you mean that the signal to use yarn instead of npm ahould be existence of that in the system then I must disagree, because it's perfectly normal to use yarn only on selected project on a developer machine and use npm on others.